app/Http/Controllers/Api/Paciente/ComplicacoesVentilacaoMec/ComplicacoesVentilacaoMecController.php

```diff
@@ -147,37 +129,20 @@ public function index($pacienteId)
      */
     public function store(ComplicacaoVentilacaoMecanicaRequest $request, $pacienteId)
     {
-        if ($request->has(['tipo_complicacao_id', 'data_complicacao'])) {
+        try {
             $complicacaoVentilacaoMec = ComplicacaoVentilacaoMec::create(array_merge(
-                $request->only(['tipo_complicacao_id', 'data_complicacao', 'descricao']),
-                [
-                    'paciente_id' => $pacienteId
-                ]
-            ));
-
-            if ($request->input('tipo_complicacao_id') == 4) {
-                $transfusaoOcorrencia = new TransfusaoOcorrencia();
-                $transfusaoOcorrencia->tipo_transfusao_id = $request->tipo_transfusao_id;
-                $transfusaoOcorrencia->data_transfusao = $request->data_complicacao;
-                $transfusaoOcorrencia->volume_transfusao = $request->volume_transfusao;
-                $transfusaoOcorrencia->paciente_id = $pacienteId;
-                $transfusaoOcorrencia->save();
-            }
-        }
-
-        if ($request->has(['tipo_transfusao_id', 'data_transfusao', 'volume_transfusao'])) {
-            $transfusaoOcorrencia = TransfusaoOcorrencia::create(array_merge(
-                $request->only(['tipo_transfusao_id', 'data_transfusao', 'volume_transfusao']),
+                $request->all(),
                 [
-                    'paciente_id' => $pacienteId
-                ]
+                'paciente_id' => $pacienteId
+            ]
             ));
-        }
 
-        return response()->json([
+            return response()->json([
             "message" => "Complicação ventilação mecânica cadastrado com sucesso",
-            "ventilacao_mecanica" => isset($complicacaoVentilacaoMec) ? $complicacaoVentilacaoMec->toArray() : [],
-            "transfusao_ocorrencia" => isset($transfusaoOcorrencia) ? $transfusaoOcorrencia->toArray() : []
+            "ventilacao_mecanica" => $complicacaoVentilacaoMec->toArray(),
         ], 201);
+        } catch (Exception $e) {
+            dd($e);
+        }
```

Comment:
Qual erro aqui poderia acontecer? Esse erro aconteceu? Este erro é possível ser contornado? Este erro deve parar o sistema?

São perguntas que eu faço quando penso em fazer um try-catch. No geral eu evito em quase 100% o uso do try-catch porque é melhor retornar um erro 500 do que tratar de forma errada.
Uma forma de se utilzar o try-catch é quando você quer fazer um rollback de alguma coisa feita antes do try, o que não é este caso.
Então eu removeria este try-catch, que simplesmente da um DD.

Reply:
Acho que se esqueceu de chamar a validação na request. Mesmo estando definida em rules através do formrequest, é preciso chamar $request->validate();

Um exemplo lá da doc.

```php
public function store(StoreBlogPost $request)
{
    // The incoming request is valid...

    // Retrieve the validated input data...
    $validated = $request->validated();
}
```

Reply:
O DD foi que esqueci quando estava corrigindo os testes.
Quanto ao validated, não está sendo utilizado nas outras rotas e retorna os erros. Mas se for necessário posso adicionar sem problemas!

Reply:
Que estranho na doc mandar uma chamada no validade e na pratica ele chama automaticamente. :D
Se ele já está chamando validade, não precisa adicionar.
Só me reconfirma isso:
Está executando o validate mesmo sem chamá-lo explicitamente??